compassion_sweden_payment/models/contract_group.py

```diff
@@ -20,14 +20,15 @@ class RecurringContract(models.Model):
 
     @api.model
     def create(self, vals):
-        group = self.env['recurring.contract.group'].browse(vals.get('group_id'))
-        if vals.get('reference', '/') == '/':
-            vals['reference'] = self.env['ir.sequence'].next_by_code(
-                'recurring.contract.ref')
-            partner_ref = self.env['res.partner'].browse(vals.get('partner_id')).ref
-            ref = f'7{int(partner_ref):05d}{int(vals["reference"][3:]):09d}'
-            check_digit = (10 - reduce(lambda a, b: (a + int(b / 10) + b),
-                                       map(lambda b: (2 if (b[0] & 1 == 0) else 1) * int(b[1]), enumerate(ref)),
-                                       0)) % 10
-            group.update({'ref': f"{ref}{check_digit}"})
+        if self.env.company.country_id == self.env.ref('base.se'):
```

Comment:
Should we not check the company from the contract instead of the environment?
```suggestion
        company_id = vals.get("company_id", self.env.company.id)
        company = self.env["res.company"].browse(company_id)
        if company.country_id == self.env.ref('base.se'):
```